satpy/readers/seviri_l1b_native.py

```diff
@@ -554,6 +571,13 @@ def _get_calib_coefs(self, channel_name):
             'RadiometricProcessing']['MPEFCalFeedback']
         radiance_types = self.header['15_DATA_HEADER']['ImageDescription'][
                 'Level15ImageProduction']['PlannedChanProcessing']
+
+        # Merge potential external and EUMETSAT climatology coefficients
+        if self.calib_mode == 'EUMCLIM':
+            tmp_dict = self._sort_ext_eum_coefs()
+        else:
+            tmp_dict = self.ext_calib_coefs
```

Comment:
Could the `self.ext_calib_coefs` be set with the sorted EUMCLIM values already in the `if self.calib_mode == 'EUMCLIM` branch in `__init__()`? Then `tmp_dict` and the if/else wouldn't be needed.